test/e2e/placement_pickn_test.go

```diff
@@ -0,0 +1,550 @@
+												MatchLabels: map[string]string{
+													regionLabelName: regionLabelValue1,
+												},
+											},
+										},
+									},
+								},
+							},
+						},
+						TopologySpreadConstraints: []placementv1beta1.TopologySpreadConstraint{
+							{
+								MaxSkew:           pointer.Int32(1),
+								TopologyKey:       envLabelName,
+								WhenUnsatisfiable: placementv1beta1.DoNotSchedule,
+							},
+						},
+					},
+					Strategy: placementv1beta1.RolloutStrategy{
+						Type: placementv1beta1.RollingUpdateRolloutStrategyType,
+						RollingUpdate: &placementv1beta1.RollingUpdateConfig{
+							UnavailablePeriodSeconds: pointer.Int(2),
+						},
+					},
+				},
+			}
+			Expect(hubClient.Create(ctx, crp)).To(Succeed(), "Failed to create CRP")
+		})
+
+		It("should update CRP status as expected", func() {
+			statusUpdatedActual := crpStatusUpdatedActual(workResourceIdentifiers(), []string{memberCluster1EastProdName, memberCluster2EastCanaryName}, nil)
+			Eventually(statusUpdatedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to update CRP status as expected")
+		})
+
+		It("should place resources on the picked clusters", func() {
+			targetClusters := []*framework.Cluster{memberCluster1EastProd, memberCluster2EastCanary}
+			for _, cluster := range targetClusters {
+				resourcePlacedActual := workNamespaceAndConfigMapPlacedOnClusterActual(cluster)
+				Eventually(resourcePlacedActual, eventuallyDuration, eventuallyInterval).Should(Succeed(), "Failed to place resources on the picked clusters")
+			}
+		})
+
+		AfterAll(func() {
+			ensureCRPAndRelatedResourcesDeletion(crpName, []*framework.Cluster{memberCluster1EastProd, memberCluster2EastCanary})
+		})
+	})
+
+	Context("affinities and topology spread constraints updated", Ordered, func() {
+		crpName := fmt.Sprintf(crpNameTemplate, GinkgoParallelProcess())
+
+		BeforeAll(func() {
+			// Create the resources.
+			createWorkResources()
+
+			// Create the CRP.
+			crp := &placementv1beta1.ClusterResourcePlacement{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: crpName,
+					// Add a custom finalizer; this would allow us to better observe
+					// the behavior of the controllers.
+					Finalizers: []string{customDeletionBlockerFinalizer},
+				},
+				Spec: placementv1beta1.ClusterResourcePlacementSpec{
+					ResourceSelectors: workResourceSelector(),
+					Policy: &placementv1beta1.PlacementPolicy{
+						PlacementType:    placementv1beta1.PickNPlacementType,
+						NumberOfClusters: pointer.Int32(2),
+						Affinity: &placementv1beta1.Affinity{
+							ClusterAffinity: &placementv1beta1.ClusterAffinity{
+								RequiredDuringSchedulingIgnoredDuringExecution: &placementv1beta1.ClusterSelector{
+									ClusterSelectorTerms: []placementv1beta1.ClusterSelectorTerm{
+										{
+											LabelSelector: metav1.LabelSelector{
+												MatchLabels: map[string]string{
+													regionLabelName: regionLabelValue1,
```

Comment:
where is the "regionLabelValue1" defined? What is its value?

Reply:
Hi Ryan! It's in the `setup_test.go` file (line 81 onwards).